lib/usd/ui/views/TreeModel.cpp

```diff
@@ -228,12 +221,77 @@ void TreeModel::openPersistentEditors(QTreeView* tv, const QModelIndex& parent)
 			QSortFilterProxyModel* proxyModel = qobject_cast<QSortFilterProxyModel*>(tv->model());
 			tv->openPersistentEditor(proxyModel->mapFromSource(varSelIndex));
 		}
-		QModelIndex childIndex = this->index(r, 0, parent);
+
+		// Note: only the load column (0) has children, so we use it when looking for children.
+		QModelIndex childIndex = this->index(r, kTreeColumn_Load, parent);
 		if (hasChildren(childIndex))
 		{
 			openPersistentEditors(tv, childIndex);
 		}
 	}
 }
 
+void TreeModel::setRootPrimPath(const std::string& path)
+{
+	// Find the prim matching the root prim path from the import data and
+	// check-enable it.
+	SdfPath rootPrimPath(path);
+	auto fnFindRoot = [rootPrimPath](TreeItem* item) -> bool
+	{
+		if (item->prim().GetPath() == rootPrimPath)
+			return true;
+		return false;
+	};
+	TreeItem* item = findTreeItem(this, QModelIndex(), fnFindRoot);
+	if (item != nullptr)
+	{
+		checkEnableItem(item);
+	}
+}
+
+void TreeModel::uncheckEnableTree()
+{
+	// When unchecking any item we uncheck-enable the entire tree.
+	setChildCheckState(QModelIndex(), TreeItem::CheckState::kUnchecked);
+}
+
+void TreeModel::checkEnableItem(TreeItem* item)
+{
+	if (item != nullptr)
+	{
+		// All ancestors [and their descendants] become unchecked-disabled, and all descendants
+		// become checked-disabled.
+
+		// First run-thru entire tree and uncheck-disable everything.
+		setChildCheckState(QModelIndex(), TreeItem::CheckState::kUnchecked_Disabled);
+
+		// Then check the item that was clicked.
+		item->setCheckState(TreeItem::CheckState::kChecked);
+		QModelIndex modelIndex = indexFromItem(item);
+		QVector<int> roles;
+		roles << Qt::DecorationRole;
+		emit dataChanged(modelIndex, modelIndex, roles);
+
+		// Then check-disable all the children of the clicked item.
+		setChildCheckState(modelIndex, TreeItem::CheckState::kChecked_Disabled);
+	}
+}
+
+void TreeModel::onItemClicked(TreeItem* item)
```

Comment:
Called by the dialog in response to clicking on an item. We use it to toggle the checkstate.